internal/revision/scanner.go

```diff
@@ -0,0 +1,124 @@
+package revision
+
+import (
+	"bufio"
+	"io"
+	"unicode"
+)
+
+var zeroRune = rune(0)
+
+// scanner represents a lexical scanner.
+type scanner struct {
+	r *bufio.Reader
+}
+
+// newScanner returns a new instance of scanner.
+func newScanner(r io.Reader) *scanner {
+	return &scanner{r: bufio.NewReader(r)}
+}
+
+// read reads the next rune from the bufferred reader.
+// Returns the rune(0) if an error occurs (or io.EOF is returned).
+func (s *scanner) read() rune {
+	ch, _, err := s.r.ReadRune()
+	if err != nil {
+		return zeroRune
+	}
+	return ch
+}
+
+// unread places the previously read rune back on the reader.
+func (s *scanner) unread() { _ = s.r.UnreadRune() }
+
+// Scan extract tokens from an input
```

Comment:
I updated the comment.

About string value it's used at several places :

https://github.com/src-d/go-git/pull/139/files#diff-baaf5fc3681a156b2bf66e6ae6a30c78R273
https://github.com/src-d/go-git/pull/139/files#diff-baaf5fc3681a156b2bf66e6ae6a30c78R275
https://github.com/src-d/go-git/pull/139/files#diff-baaf5fc3681a156b2bf66e6ae6a30c78R309
https://github.com/src-d/go-git/pull/139/files#diff-baaf5fc3681a156b2bf66e6ae6a30c78R364
and so on